### ISSUE TITLE
URL Validator does not allow pipes

### ISSUE THREAD

**Issue by VinceG** [laravel/framework#33887]
- Laravel Version: 7.x
- PHP Version: 7.4.5
- Database Driver & Version: MySQL 5.7 (irrelevant)

### Description:

For some reason in Laravel 7.x the 'url' validator does not allow pipes (|) while Laravel 6.x does.

### Steps To Reproduce:

Using Laravel 7.x run the following code:

```
$url = "https://laravel.com?foo=one|two";
$validator = Validator::make(
    [
    	'url' => $url
    ],
    [
        'url' => 'url'
    ]
);
$validator->errors()->toArray();
```

<img width="938" alt="Screen Shot 2020-08-14 at 4 47 45 PM" src="https://user-images.githubusercontent.com/195199/90300193-3c609780-de4e-11ea-8721-907aca1d186d.png">

<img width="1276" alt="Screen Shot 2020-08-14 at 4 46 03 PM" src="https://user-images.githubusercontent.com/195199/90300197-3ec2f180-de4e-11ea-8e28-46d9e5cc25bf.png">

**Comment by Micky5991** [comment]
I also came across this error when I try to validate an urlencoded url with the same pipe:

`%7C` = `|`

**URL**
```
https://laravel.com/somefile%7Cothername.mp4
```

**Validation error in json**
```json
{
    "message": "The given data was invalid.",
    "errors": {
        "url": [
            "The url format is invalid."
        ]
    }
}
```

**Comment by rbarden** [comment]
It looks like the URL validation code was changed in pull request #32315 . I'm not any good with regex, but the answer may lie in there.

**Comment by VinceG** [comment]
@rbarden Yep, saw that while I was backtracking the change. Since this comes from Symfony it might be more than just us changing the regex. I mean I can submit a PR to add the pipe to the regex, I just want to see if the team here will accept it.

**Comment by rbarden** [comment]
You should submit the PR and they'll either accept or reject it. Make sure to give context.

**Comment by VinceG** [comment]
@rbarden I rather wait to get confirmation if they'll accept a PR for this change first since in the past ([#17908](https://github.com/laravel/framework/pull/17908), [#379](https://github.com/laravel/horizon/pull/379), [#380](https://github.com/laravel/horizon/pull/380)) I've spent the time creating the PR's and they were closed.

**Comment by taylorotwell** [comment]
We usually just port this regular expression from Symfony. I would see if ours matches theirs. If it does, you may want to submit a PR to them to fix it on their end and then we can port it over.
